src/dahua.ts

```diff
@@ -28,51 +27,57 @@ class DahuaEvents {
     public RECONNECTING_EVENT_NAME: string = 'reconnecting'
 
     private host:                   string
+    private user:                   string
+    private pass:                   string
+    private requestConfig:          AxiosRequestConfig
 
     constructor(host: string, user: string, pass: string, useHttp: boolean) {
         this.host = host
-        const auth: AxiosBasicCredentials = {
-            username: user,
-            password: pass
-        }
+        this.user = user
+        this.pass = pass
 
-        let keepAliveAgent
-        if(useHttp) {
-            keepAliveAgent = new HttpAgent({
+        const keepAliveAgent: HttpAgent | HttpsAgent = useHttp ?
+            new HttpAgent({
                 keepAlive: this.AGENT_SETTINGS.keepAlive,
                 keepAliveMsecs: this.AGENT_SETTINGS.keepAliveMsecs,
                 maxSockets: this.AGENT_SETTINGS.maxSockets,
                 maxFreeSockets: this.AGENT_SETTINGS.maxFreeSockets,
-            }) 
-        } else {
-            keepAliveAgent = new HttpsAgent({
+            })
+            :
+            new HttpsAgent({
                 keepAlive: this.AGENT_SETTINGS.keepAlive,
                 keepAliveMsecs: this.AGENT_SETTINGS.keepAliveMsecs,
                 maxSockets: this.AGENT_SETTINGS.maxSockets,
                 maxFreeSockets: this.AGENT_SETTINGS.maxFreeSockets,
                 rejectUnauthorized: false,
                 minVersion: "TLSv1"
             })
-        }
 
-        let useSSL = useHttp ? 'http': 'https'
-        const axiosRequestConfig: AxiosRequestConfig ={
-            url: `${useSSL}://${host}${this.EVENTS_URI}`,
+        this.requestConfig = {
+            url: `${useHttp ? 'http': 'https'}://${host}${this.EVENTS_URI}`,
             httpsAgent: keepAliveAgent, 
             httpAgent: keepAliveAgent,
-            auth: auth,
-            headers: this.HEADERS,
             method: 'GET',
-            responseType: 'stream'
+            responseType: 'stream',
+
         }
 
         this.eventEmitter = new EventEmitter()
 
-        this.connect(axiosRequestConfig, 0)
+        this.connect(0)
     }
 
-    private connect = (axiosRequestConfig: AxiosRequestConfig, count: number) => {
-        Axios.request(axiosRequestConfig).then((res: AxiosResponse) => {
+    private connect = (count: number, authHeader?: string) => {
+        Axios.request({
+            ...this.requestConfig, 
+            headers: {
+                'accept': 'multipart/x-mixed-replace',
+                'authorization': authHeader ?
+                    authHeader :
+                    // basic auth
+                    Buffer.from(`${this.user}:${this.pass}`).toString('base64')
+            }
+        }).then((res) => {

```

Comment:
```suggestion
        }).then((res: AxiosResponse) => {

```

This response type should be kept

Reply:
Will add back but I think the inferred response type is the same so it's redundant?

<img width="493" alt="image" src="https://user-images.githubusercontent.com/484912/205013900-95d445fa-c3b6-425c-82d5-da4b15bc6d3d.png">

Reply:
You're totally right, thanks for changing it anyhow.